pingcap/tidb-operator
# Add util class to support to add annotations to Grafana

**Function**
1. Support to add annotations to Grafana
2. Add a errorCount metrics to record how many error occurred.
Prometheus can scrape this metric by xxx:8083/metrics.

**Usage**
1. `cli, _ := NewGrafanaClient("http://$GrafanaIP:3000", "$GrafanaUserName", "$GrafanaPassword", "$PrometheusScrapePort")`
2. `annotation := Annotation{
	dashboardId: $dashboardId,
	panelId: $panelId,
	tags: []string{"tag"},
	timestampInMilliSec: time.Now().Unix() * 1000,
	text: "test",
	}`
3. `cli.AddAnnotation(annotation)`
4.  `cli.IncrErrorCount()`

**Test**
1. Add Unit Test Case - PASS
2. Check the Grafana Annotation - PASS
![image](https://user-images.githubusercontent.com/7285248/55801211-24655f00-5b08-11e9-8377-2f416317a5b1.png)
3. Check the metrics by prometheus exporter - PASS
![image](https://user-images.githubusercontent.com/7285248/55801269-44951e00-5b08-11e9-97b2-eb9a97aaa4e6.png)
